tools/clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -266,6 +267,24 @@ void Sema::InstantiateAttrs(const MultiLevelTemplateArgumentList &TemplateArgs,
       continue;
     }
 
+    // HLSL Change Begin - Validate post-instantiation attributes
+    if (const HLSLGloballyCoherentAttr *HLSLGCAttr =
+            dyn_cast<HLSLGloballyCoherentAttr>(TmplAttr)) {
+      ValueDecl *TD = dyn_cast<ValueDecl>(New);
+      if (TD && !TD->getType()->isDependentType()) {
```

Comment:
Is it possible to share code with handleHLSLGloballyCoherent? The code feels very similar.

Reply:
I commented above that I was cleaning up the code, that will be reworked in my update.